src/globus_sdk/experimental/session_error/session_error.py

```diff
@@ -0,0 +1,195 @@
+import copy
+import typing as t
+
+from globus_sdk.exc import GlobusError
+
+
+class GlobusSessionErrorAuthorizationParameters:
+    """
+    Represents authorization parameters that can be used to instruct a client
+    which additional authorizations are needed in order to complete a request.
+
+    :ivar session_message: A message to be displayed to the user.
+    :vartype session_message: str, optional
+
+    :ivar session_required_identities: A list of identities required for the
+        session.
+    :vartype session_required_identities: list of str, optional
+
+    :ivar session_required_policies: A list of policies required for the
+        session.
+    :vartype session_required_policies: list of str, optional
+
+    :ivar session_required_single_domain: A list of domains required for the
+        session.
+    :vartype session_required_single_domain: list of str, optional
+
+    :ivar session_required_mfa: Whether MFA is required for the session.
+    :vartype session_required_mfa: bool, optional
+
+    :ivar session_required_scopes: A list of scopes for which consent is required.
+    :vartype session_required_scopes: list of str, optional
+
+    :ivar extra_fields: A dictionary of additional fields that were provided. May
+        be used for forward/backward compatibility.
+    :vartype extra_fields: dict
+    """
+
+    SUPPORTED_FIELDS = {
+        "session_message": str,
+        "session_required_identities": list,
+        "session_required_policies": list,
+        "session_required_single_domain": list,
+        "session_required_mfa": bool,
+        "session_required_scopes": list,
+    }
```

Comment:
I know it's tempting to build something which looks pretty and generic, but I think this loses valuable precision. Specifically: `list[str]` can't be validated fully (as containing only strings) based on this logic.

I think we can build ourselves something only _slightly_ less pretty and get a good result. What about the following alternative structure?
```python
# imagine `globus_sdk.experimental.session_error.validators` is a module
_FIELD_VALIDATORS = {
    "session_message": validators.opt_str,
    "session_required_identities": validators.opt_str_list,
    ...
}
```